### PR TITLE
Add install.sh to automate installation for developing under Linux

### PR DESCRIPTION
**Pull request by feinstaub**
Plus: Fix errors in nxt_test and add a question.

### CODE REVIEW

**Comment by Eelviny** [comment]
Thanks for the updates. I was literally typing out what `if b in locals` did when you submitted a commit fixing it! You found a section that escaped my notice when I converted the codebase from Python 2. 

That section is needed to cleanly disconnect the NXT in the event of the tests failing - up until now it didn't account for not being able to connect to the NXT at all, which lead to the variable `b` not being assigned.

Your additions are very helpful - wish I had time for programming the thing these days, but I always leave time to merge pull requests.

**Comment by feinstaub** [comment]
Thanks for taking your time to review these things.

**Comment by Eelviny** [comment]
No worries, I'm a student in the run up to exams right now, but I'll do my best.

**Comment by feinstaub** [comment]
Best time to be susceptible to do some coding/reviews. ;-)

**Comment by feinstaub** [comment]
Off-topic: when you are working with git, which tools do you use? (I recently switched from Windows where I used GitExtensions to Linux and my current workflow feels kind of slow)

**Comment by Eelviny** [comment]
@feinstaub 

- gitg is a graphical tool I use for visualising the repository and creating commits. Great for simple Git usage, but is more of specific tool than a full-feature git management suite.
- The rest goes through command line. Pushing and pulling to Github goes through this, along with much more complicated aspects such as reverts and rebases. It's the most powerful method of git management, albeit the most complicated!

When it comes to actually programming, I actually use gedit for _everything._ Gedit is highly extensible, and it's a breeze to add plugins such as line numbers, consoles and - most the most useful - a git line change indicator, showing green, orange or red depending on which lines I've changed since the last commit.

File: scripts/nxt_test [hunk 1]
```diff
@@ -21,13 +22,13 @@ try:
...
+    print('Firmware version %s.%s' % fw_version)
     millivolts = b.get_battery_level()
     print('Battery level %s mV' % millivolts)
     b.sock.close()
 except:
     print("Error while running test:")
     traceback.print_tb(sys.exc_info()[2])
-    print str(sys.exc_info()[1])
+    print(str(sys.exc_info()[1]))
     if b in locals():
```
Comment (merwok) [inline scripts/nxt_test]: locals returns a dict that represents the local namespace; the keys are strings, so to fix this code you need to have either: `if 'b' in locals()` or `if b is not None`.

**Pull request merged**